### ISSUE TITLE
[Jaeger v2] Add Kafka exporter and receiver

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger#4868]
Part of #4843

## Objective
  * Demonstrate using Kafka exporter & receiver from OTEL Collector Contrib with `jaeger-v2` as a replacement for `jaeger-collector` / `jaeger-ingester`
  * Document gaps in backwards compatibility

## Design

We want to rely on OTel's usual pipeline setup, which should allow us to use OTel's Kafka receiver and exporter without changes. For example, this is how it might look

### Replacement for jaeger-collector
```shell
jaeger-v2 --config collector-with-kafka.yaml
```

```yaml
# collector-with-kafka.yaml
service:
  # extensions: [jaeger_storage, jaeger_query] # don't really need these for collector with Kafka
  pipelines:
    traces:
      receivers: [otlp, jaeger, zipkin]
      processors: [batch]
      exporters: [kafka]  # using standard Kafka exporter as the only one
```

### Replacement for jaeger-ingester
```shell
jaeger-v2 --config ingester.yaml
```

```yaml
# ingester.yaml
service:
  extensions: [jaeger_storage, jaeger_query]
  pipelines:
    traces:
      receivers: [kafka] # using standard Kafka receiver as the only one
      processors: [batch]
      exporters: [jaeger_storage_exporter]  # same as in the default cmd/jaeger-v2/config.yaml
```

## Steps
  * [ ] Add wiring and sample configuration to prototype #4766
  * [ ] Test backwards compatibility:
    * [ ] Sending spans in Jaeger format to Kafka
    * [ ] Receiving spans in all formats that `jaeger-ingester` supports today
  * [ ] Document gaps
  * [ ] Provide reproducible integration test scripts (manually run)

This issue is not a one-evening-worth kind, but the steps are simple enough that I think it can be `good-first-issue`.

**Comment by NavinShrinivas** [comment]
Hey, I'd like to try this 

Just to warrant my understanding : 
- First include `Kafka` configurations in the config files ( receiver : https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kafkareceiver/README.md and exporter : https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/kafkaexporter/README.md)
- I'm assuming here that the single v2 binary is responsible for ingesting and querying this means : 
   - kafka recvier will get traces (in jeager format from various data sources)
   - Will export data to various stores(using kafka)? [Im Not sure about this point] OR are we testing is otel exporters can export jaeger traces to kafka?

**Comment by yurishkuro** [comment]
> I'm assuming here that the single v2 binary is responsible for ingesting and querying this means 

yes but it can be configured to run in different roles

> Will export data to various stores(using kafka)? [Im Not sure about this point] OR are we testing if otel exporters can export jaeger traces to kafka?

the "collector" mode will be receiving spans over network, e.g. OTLP receiver, and using Kafka exporter to write them to Kafka. The "ingester" mode will be using Kafka receiver and then using Jaeger storage exporter to persist in a database.

**Comment by NavinShrinivas** [comment]
We only seem to be providing one config.yaml (which I assume includes config for both collector and ingester, as mentioned in the doc [Option 2 (selected): shared runtime component] ) So the config will look something like this : 
On collector end : 
```
extensions:
  jaeger_storage:
    kafka:
        kafka_traces:
             kafka_topic: traces

exporters:
  jaeger_storage_exporter:
    trace_storage: kafka_traces
```
I'm still not sure how the "ingester" (Which I think is the "jeager_storage" is able to reference another Jaeger storage (Kafka). Also, are these features even supported and implemented in the v2 binary (To read something like the above config file)?

**Comment by varshith257** [comment]
@yurishkuro Is this WIP or needed to work on for those not has any merged PRs:

- [ ] Test backwards compatibility:
- [ ]  Sending spans in Jaeger format to Kafka
- [ ]  Receiving spans in all formats that jaeger-ingester supports today
- [ ]  Document gaps
- [ ]  Provide reproducible integration test scripts (manually run)

**Comment by yurishkuro** [comment]
@james-ryans has it

**Comment by varshith257** [comment]
@james-ryans Any updates here?

**Comment by james-ryans** [comment]
> @james-ryans Any updates here?

The PR addressing this issue is #4971 and is currently in progress but is blocked by an upstream issue at [otelcol-contrib#30789](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30789). I haven't received enough context to assist in fixing the upstream issue, as we have prioritized adding full support for Jaeger-v2 storage backends. I'll return to this once our priorities have been addressed.
